skia/RSkComponentRNSVGUse.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 1994-2023 OpenTV, Inc. and Nagravision S.A.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "RSkComponentRNSVGUse.h"
+
+namespace facebook {
+namespace react {
+
+RSkComponentRNSVGUse::RSkComponentRNSVGUse(const ShadowView &shadowView)
+    : RSkComponent(shadowView,LAYER_TYPE_VIRTUAL),
+      INHERITED(SkSVGTag::kUse) {}
+
+RnsShell::LayerInvalidateMask  RSkComponentRNSVGUse::updateComponentProps(SharedProps newViewProps,bool forceUpdate) {
+  auto component = getComponentData();
+
+  auto const &newRNSVGUseProps = *std::static_pointer_cast<RNSVGUseProps const>(newViewProps);
+  
+  RNS_LOG_DEBUG( " Width :: "<<component.layoutMetrics.frame.size.width<<" Height :: "<<component.layoutMetrics.frame.size.height<< " X:: "<<component.layoutMetrics.frame.origin.x<< " Y:: "<<component.layoutMetrics.frame.origin.y);
+  setNativeProps(newRNSVGUseProps);
+  setCommonRenderableProps(newRNSVGUseProps);
+  setCommonNodeProps(newRNSVGUseProps);
+ 
+  return RnsShell::LayerInvalidateNone;
+}
+
+inline RnsShell::LayerInvalidateMask  RSkComponentRNSVGUse::setNativeProps(const RNSVGUseProps &nativeProps) {
+
+  RNS_LOG_INFO(" X: "<<nativeProps.x);
+  RNS_LOG_INFO(" Y: "<<nativeProps.y);
+  RNS_LOG_INFO(" Href: "<<nativeProps.href.c_str());
+  RNS_LOG_INFO(" Width: "<<nativeProps.width);
+  RNS_LOG_INFO(" Height: "<<nativeProps.height);
+
+  setLengthAttribute(SkSVGAttribute::kX,nativeProps.x.c_str());
+  setLengthAttribute(SkSVGAttribute::kY,nativeProps.y.c_str());
+  setAttribute(SkSVGAttribute::kHref,SkSVGStringValue(SkString(nativeProps.href.c_str())));
+  setLengthAttribute(SkSVGAttribute::kWidth,nativeProps.width.c_str());
+  setLengthAttribute(SkSVGAttribute::kHeight,nativeProps.height.c_str());
+
+  return RnsShell::LayerInvalidateNone;
```

Comment:
should this be layoutInvalidate ?

Reply:
You are right, It have to be Layout Invalidate, will correct it.